<!DOCTYPE html>
<html>
   <head>
      <title>flake8 violations: .venv/lib/python3.13/site-packages/pandas/errors/__init__.py</title>
      <meta http-equiv="Content-Type" value="text/html; charset=UTF-8">
      <link rel="stylesheet" href="styles.css">
<script>
function bindHandler(link) {
   var code = link.getAttribute('data-code');
   var key = location.pathname + '#' + code;

   var ul = link.parentNode.querySelector('.details');
   if (sessionStorage[key] != 'open') {
      ul.style.display = 'none';
   }
   link.addEventListener('click', function (event) {
      if (!ul.style.display || ul.style.display == 'none') {
         ul.style.display = 'block';
         sessionStorage[key] = 'open';
      } else {
         ul.style.display = 'none';
         sessionStorage[key] = 'closed';
      }
   });
}

window.addEventListener('DOMContentLoaded', function () {
   var links = document.querySelectorAll('#index > li > a');
   for (var i = 0; i < links.length; i++) {
      bindHandler(links[i]);
   }
});
</script>
   </head>
   <body>
      <div id="masthead" class="sev-2"></div>
      <div id="page">
         <p id="srclink">
            <a title="View full annotated source"
               href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html">
               <img src="file.svg" alt="&#x2261;">
            </a></p>
         <h1>
            <a href="index.html">
               <img src="back.svg" alt="&#x2B05;">
               .venv/lib/python3.13/site-packages/pandas/errors/__init__.py
            </a>
         </h1>

         <ul id="index">
         
         <li>
            <a data-code="E501">
               <span class="count sev-2">
                  36
               </span>
               <strong>E501:</strong> line too long (87 > 79 characters)
               
                  (and 8 similar)
               
            </a>
            <ul class="details">
            <li>
               
               <p>Line too long (87 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l20">
                  <tt><i>20</i> <span class="sd">    Exception raised when converting (``astype``) an array with NaN to an integer type.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l192">
                  <tt><i>192</i> <span class="sd">    &gt;&gt;&gt; df2 = pd.read_csv(&#39;test.csv&#39;, sep=&#39;,&#39;, dtype={&#39;a&#39;: str})  # doctest: +SKIP</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (81 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l200">
                  <tt><i>200</i> <span class="sd">    Exception raised in ``pd.read_csv`` when empty data or header is encountered.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (81 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l269">
                  <tt><i>269</i> <span class="sd">    MergeError: Merge keys are not unique in left dataset; not a one-to-one merge</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l282">
                  <tt><i>282</i> <span class="sd">    ...         raise pd.errors.AbstractMethodError(cls, methodtype=&quot;classmethod&quot;)</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (83 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l287">
                  <tt><i>287</i> <span class="sd">    AbstractMethodError: This classmethod must be defined in the concrete class Foo</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (83 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l291">
                  <tt><i>291</i> <span class="sd">    AbstractMethodError: This classmethod must be defined in the concrete class Foo</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (85 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l308">
                  <tt><i>308</i>         <span class="k">return</span> <span class="sa">f</span><span class="s2">&quot;This </span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">methodtype</span><span class="si">}</span><span class="s2"> must be defined in the concrete class </span><span class="si">{</span><span class="n">name</span><span class="si">}</span><span class="s2">&quot;</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (87 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l515">
                  <tt><i>515</i>     <span class="s2">&quot;assignment in a single step and ensure this keeps updating the original `df`.</span><span class="se">\n\n</span><span class="s2">&quot;</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l536">
                  <tt><i>536</i>     <span class="c1"># This is a mess, selection paths that return a view set the _cacher attribute</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (83 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l537">
                  <tt><i>537</i>     <span class="c1"># on the Series; most of them also set _item_cache which adds 1 to our relevant</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (80 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l538">
                  <tt><i>538</i>     <span class="c1"># reference count, but iloc does not, so we have to check if we are actually</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (83 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l547">
                  <tt><i>547</i>                 <span class="c1"># Check if we are actually the item from item_cache, iloc creates a</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (83 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l555">
                  <tt><i>555</i> <span class="sd">    Exception raised when trying to use a built-in numexpr name as a variable name.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (80 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l558">
                  <tt><i>558</i> <span class="sd">    to &#39;numexpr&#39;. &#39;numexpr&#39; is the default engine value for these methods if the</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l565">
                  <tt><i>565</i> <span class="sd">    ... # NumExprClobberingError: Variables in expression &quot;(abs) &gt; (2)&quot; overlap...</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l568">
                  <tt><i>568</i> <span class="sd">    ... # NumExprClobberingError: Variables in expression &quot;(sin) + (a)&quot; overlap...</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (84 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l574">
                  <tt><i>574</i> <span class="sd">    Exception raised by ``query`` or ``eval`` when using an undefined variable name.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (83 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l600">
                  <tt><i>600</i> <span class="sd">    Exception is raised when trying to index and there is a mismatch in dimensions.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (80 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l645">
                  <tt><i>645</i> <span class="sd">    This can be due to the styling not having an equivalent value or because the</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (80 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l670">
                  <tt><i>670</i> <span class="sd">    ... # PossibleDataLossError: Re-opening the file [my-store] with mode [a]...</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (86 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l676">
                  <tt><i>676</i> <span class="sd">    Exception is raised when trying to perform an operation on a closed HDFStore file.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l689">
                  <tt><i>689</i> <span class="sd">    Warning raised when trying to use where criteria on an incompatible HDF5 file.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (87 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l698">
                  <tt><i>698</i> <span class="sd">    name than the existing index on an HDFStore or attempting to append an index with a</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (81 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l709">
                  <tt><i>709</i> <span class="sd">    AttributeConflictWarning: the [index_name] attribute of the existing index is</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (83 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l716">
                  <tt><i>716</i> <span class="sd">    Error is raised when executing sql with bad syntax or sql that throws an error.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l723">
                  <tt><i>723</i> <span class="sd">    ... # DatabaseError: Execution failed on sql &#39;test&#39;: near &quot;test&quot;: syntax error</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (82 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l729">
                  <tt><i>729</i> <span class="sd">    Warning raised by to_stata on a column with a value outside or equal to int64.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (84 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l744">
                  <tt><i>744</i> <span class="sd">    Warning raised by to_stata on a category column that contains non-string values.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (80 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l748">
                  <tt><i>748</i> <span class="sd">    &gt;&gt;&gt; df = pd.DataFrame({&quot;categories&quot;: pd.Series([&quot;a&quot;, 2], dtype=&quot;category&quot;)})</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (87 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l750">
                  <tt><i>750</i> <span class="sd">    ... # ValueLabelTypeMismatch: Stata value labels (pandas categories) must be str...</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (85 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l765">
                  <tt><i>765</i> <span class="sd">    ... # InvalidColumnName: Not all pandas column names were valid Stata variable...</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (81 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l771">
                  <tt><i>771</i> <span class="sd">    Warning is raised when reading a partial labeled Stata file using a iterator.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (81 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l785">
                  <tt><i>785</i> <span class="sd">    Raised when trying to do a __setitem__ on an np.ndarray that is not lossless.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (88 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l795">
                  <tt><i>795</i> <span class="sd">    Exception is raised in _get_data_buffer to signal that there is no requested buffer.</span></tt>
               </a>
            </li><li>
               
               <p>Line too long (88 > 79 characters):</p>
               
               <a href="venv.lib.python3.13.site-packages.pandas.errors.__init__.source.html#l801">
                  <tt><i>801</i> <span class="sd">    Exception is raised by _validate_comparison_value to indicate an invalid comparison.</span></tt>
               </a>
            </li>
            </ul>
         </li>
         
         </ul>
      </div>
   </body>
</html>